CMSgov/design-system
# [WNMGDS-658] Move grid styles to utilities

Grid classes should work like utilities, where they take higher precedence than components or other classes.

This PR:
- re-group the grid classes in the utilities folder
- Move the grid doc page into utilities as well, similar to USWDS, show at the top of the utilities navigation. 
- Rewrite the layout grid documentation to be more consistent with utility doc page conventions.
- Fixes Github issue: https://github.com/CMSgov/design-system/issues/781

Note: Mgov design system
- To adopt this PR, remove `@import "@cmsgov/design-system/dist/scss/base/grid.scss";` line from file `src/styles/index.scss`

### How to test
- Run the doc site locally with `yarn start`
- Check that `Layout grid` option is grouped at the top of the `Utilities` navigation section, instead of the `Styles` section
- Check the Grid Utility page content is correct
- Check the links/references to Grid utility page is correct
- Test with the "Re-order SCSS imports" PRs on [Hcgov DS](https://github.com/CMSgov/hcgov-design-system/pull/10) and [Mgov DS](https://github.cms.gov/MedicareGov/mgov-design-system/pull/98)
- Check that github issue https://github.com/CMSgov/design-system/issues/781 is fixed
